jira/jira.go

```diff
@@ -0,0 +1,116 @@
+package jira
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+type Client struct {
+	client  *http.Client
+	baseURL *url.URL
+}
+
+func NewClient(httpClient *http.Client, baseURL string) (*Client, error) {
+	parsedBaseURL, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	if !strings.HasSuffix(baseURL, "/") {
+		baseURL += "/"
+	}
+
+	c := &Client{client: httpClient, baseURL: parsedBaseURL}
+	return c, nil
+}
+
+func (c *Client) NewRequest(method string, endpoint string, body interface{}) (*http.Request, error) {
+	parsedEndpoint, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	// Relative URLs should be specified without a preceding slash since baseURL will have the trailing slash
+	parsedEndpoint.Path = strings.TrimLeft(parsedEndpoint.Path, "/")
+
+	url := c.baseURL.ResolveReference(parsedEndpoint)
+
+	// Set request body
+	var buf io.ReadWriter
+	if body != nil {
+		buf = new(bytes.Buffer)
+		err = json.NewEncoder(buf).Encode(body)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	req, err := http.NewRequest(method, url.String(), buf)
+	if err != nil {
+		return nil, err
+	}
+	// We are only working with JSON
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+func (c *Client) UpdateIssue(issueId string, data map[string]interface{}) (*http.Response, error) {
+	endpoint := fmt.Sprintf("/rest/api/2/issue/%s", issueId)
+	req, err := c.NewRequest("PUT", endpoint, data)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func (c *Client) UpdateIssueFixVersions(tasks []string) {
```

Comment:
Similar to `UpdateIssue`, this method should return an error object, which the caller will be responsible to handle.